libp2p/Host.cpp

```diff
@@ -740,16 +764,24 @@ void Host::startedWorking()
     for (auto const& h: m_capabilities)
         h.second->onStarting();
     
-    // try to open acceptor (todo: ipv6)
-    int port = Network::tcp4Listen(m_tcp4Acceptor, m_netConfig);
-    if (port > 0)
+    if (haveCapabilities())
     {
-        m_listenPort = port;
-        determinePublic();
-        runAcceptor();
+        // try to open acceptor (todo: ipv6)
+        int port = Network::tcp4Listen(m_tcp4Acceptor, m_netConfig);
+        if (port > 0)
+        {
+            m_listenPort = port;
+            determinePublic();
```

Comment:
Why do we do this only if acceptor successfully started? `NodeTable` might need it anyway

Reply:
`m_tcpPublic` is used in `runAcceptor` only for logging, so I think we can move `determinePublic()` to just before `NodeTable` construction

Reply:
Don't we still want to log the correct public IP information in runAcceptor? Moving determinePublic to just before the NodeTable ctor will break this. I'd rather keep the correct logging at the expense of some slightly duplicated code.